src/annotator/plugin/pdf.js

```diff
@@ -164,11 +137,21 @@ export default class PDF extends Delegator {
       this._warningBanner.style.top = -rect.height + 'px';
     };
 
-    this._warningBanner = document.createElement('div');
-    this._warningBanner.className =
-      'annotator-pdf-warning-banner annotator-pdf-warning-banner--notice';
+    this._warningBanner = document.createElement('hypothesis-banner');
+    Object.assign(this._warningBanner.style, {
+      // Position the banner at the top of the viewer and make it span the full width.
+      position: 'absolute',
+      left: '0',
+      right: '0',
+
+      // The Z-index is necessary to raise the banner above the toolbar at the
+      // top of the PDF.js viewer.
+      zIndex: '10000',
```

Comment:
These styles used to be applied by the `.annotator-pdf-warning-banner` styles in `annotator.scss`. Since we dynamically adjust the `top` style of `_warningBanner` above, it is necessary to apply these styles to the `_warningBanner` container rather than its content (now in Shadow DOM). Applying this via inline styles was the easiest thing to do, although we could potentially move this to Shadow DOM in future as well, with some changes to the code above this that sets `this._warningBanner.style.top`.

Reply:
I suggests a different DOM structure that does not need the `top` value. Instead of appending to `#mainContainer` prepend to the body:

```
    this._warningBanner = document.createElement('hypothesis-banner');
    const warningBannerContent = createShadowRoot(this._warningBanner);
    render(<WarningBanner />, warningBannerContent);
    document.body.prepend(this._warningBanner);
```